src/app/ipfs-cache.ts

```diff
@@ -67,7 +132,7 @@ export class IPFSCache {
    * @param      {string}  hash    ipfs hash to store the data for
    * @param      {any}     data    data to store for the ipfs hash
    */
-  async add(hash: string, data: any) {
+  async add(hash: string, data: any): Promise<any> {
```

Comment:
I think set() has void return type.

```suggestion
  async add(hash: string, data: any): Promise<void> {
```